olive/passes/onnx/vitis_ai/quant_utils.py

```diff
@@ -238,31 +238,36 @@ def compute_scale_zp_pof2s(rmin, rmax, qmin, qmax, symmetric=False):
     # Adjust rmin and rmax such that 0 is included in the range. This is
     # required to make sure zero can be represented by the quantization data
     # type (i.e. to make sure qmin <= zero_point <= qmax)
-    rmin = min(rmin, 0)
-    rmax = max(rmax, 0)
+    rmin = np.minimum(rmin, np.array(0, dtype=rmin.dtype))
+    rmax = np.maximum(rmax, np.array(0, dtype=rmax.dtype))
 
     # Ensure that rmax-rmin is less than or equal to sys.float_info.max
     if rmin == float("-inf"):
-        rmin = -sys.float_info.max / 2
+        rmin = -np.finfo(rmin.dtype).max / 2
     if rmax == float("inf"):
-        rmax = sys.float_info.max / 2
+        rmax = np.finfo(rmax.dtype).max / 2
 
     if symmetric:
-        absmax = max(abs(rmin), abs(rmax))
+        absmax = np.maximum(np.abs(rmin), np.abs(rmax))
         rmin = -absmax
         rmax = +absmax
 
-    scale = (rmax - rmin) / float(qmax - qmin)
+    assert qmin <= qmax, f"qmin={rmin} > qmax={rmax}"
+    dr = np.array(rmax - rmin, dtype=np.float64)
+    dq = np.array(qmax, dtype=np.float64) - np.array(qmin, dtype=np.float64)
+    scale = np.array(dr / dq)
+
     pos = scale2pos(scale)
     pof2_scale = pos2scale(pos)
 
-    if pof2_scale < np.finfo(np.float32).tiny:
-        pof2_scale = 1.0
-        zero_point = 0
+    if pof2_scale < np.finfo(rmax.dtype).tiny:
+        pof2_scale = np.array(1.0, dtype=rmax.dtype)
+        zero_point = np.array(0, dtype=qmin.dtype)
     else:
-        zero_point = round(qmin - rmin / pof2_scale)
+        pof2_scale = np.array(pof2_scale, dtype=rmin.dtype)
+        zero_point = np.array(round(qmin - rmin / pof2_scale), dtype=qmin.dtype)
```

Comment:
the CI complains
![image](https://github.com/microsoft/Olive/assets/817030/7ef87891-93d8-41dd-a032-02134f136342)